### PR TITLE
Use reference labels instead of paths

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues
towards #296

# Description
 Uses references labels instead of paths. A few notes:

* I have left all the paths inside `docs/plugins/virtual_environment_docs` as they just reference files within that folder so I think we're safe
* I did not know how to deal with `gallery/` links. Does sphinx-tag automatically generate the gallery index/readme page..?
* Standardised use of `-` between words in reference labels
* Removed `napari` from some reference labels, as there is no 'plugin' equivalent thus its obvs it is about napari
* Images! - We still have a lot of relative paths to image/animation files. These would ideally be absolute paths, meaning they would be fine as long as the images stay in the same folder, but I could not be bothered. Maybe will leave this to be fixed when needed....

### CODE REVIEW

**Comment by lucyleeow** [comment]
Sorry it's not going to be fun to review. I also sort of regret taking this on :grimacing:

**Comment by psobolewskiPhD** [comment]
Yea 😅 
I think the first goal should be making sure every page has a tag/anchor that can be used for cross referencing.
Then second pass is to replace any path based links?

**Comment by lucyleeow** [comment]
Do you think thats going to be much easier to review than this PR?

**Comment by psobolewskiPhD** [comment]
I have no idea 😆 
Don't worry about it, you're doing the hard work!

File: docs/community/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ examples, you are welcome to help develop and improve napari.
 
 To ask questions and interact with the team, you can join our
 [zulip chat](https://napari.zulipchat.com/login/) or participate in our
-[meetings](./meeting_schedule).
+[meetings](meeting-schedule).
 
 If you are interested in contributing, check out our
-[contributing guide](../developers/index).
+[contributing guide](contributing).
```
Comment (psobolewskiPhD) [inline docs/community/index.md:13]: I don't get why the link works in the artifact, I'd think that this tag needs to be added to https://github.com/napari/docs/tree/main/docs/developers/index.md

File: docs/guides/contexts_expressions.md [hunk 2]
```diff
@@ -396,7 +396,7 @@ Out[12]: 1
    context menu might update the items in the menu that are visible and/or enabled:
 
    ```python
-   context_menu.update_from_context(get_context(layer_list))
+   context_menu.update_from_context(get_context(layer-list))
```
Comment (psobolewskiPhD) [inline docs/guides/contexts_expressions.md]: Is this correct? doesn't look pythonic. Also not related to this PR...

File: docs/guides/layers.md [hunk 3]
```diff
@@ -22,10 +22,10 @@ viewer.add_image(data, name='astronaut')
 ```
 
 To learn more about the layers available, see the
-[Layers documentation](napari.layers). To learn about how to use the layers currently supported by napari, check out the
-[Using layers how-to guides](../../howtos/layers/index). For a gentle
+{class}`~napari.layers`. To learn about how to use the layers currently supported by napari, check out the
```
Comment (psobolewskiPhD) [inline docs/guides/layers.md]: Can we make this a regular link like in the original? it looks wierd:
image

File: docs/guides/contexts_expressions.md (hunk 2 above)
Comment (psobolewskiPhD) [reply docs/guides/contexts_expressions.md]: Oh i bet it was search-replace!

File: docs/tutorials/fundamentals/quick_start.md [hunk 4]
```diff
@@ -22,7 +22,7 @@ napari is a fast, interactive, multi-dimensional image viewer, with [a vibrant p
 
 napari is an open source project on [GitHub](https://github.com/napari/napari) to facilitate transparency, reuse, and extensibility.
 
-At its core, it provides critical viewer features out-of-the-box, such as support for [large multi-dimensional data](../processing/dask); [“layers”](..//start_index) to simultaneously visualize images, models, and analysis results; and easy manual, interactive annotation in 3D.
+At its core, it provides critical viewer features out-of-the-box, such as support for [large multi-dimensional data](dask); [“layers”](getting-started) to simultaneously visualize images, models, and analysis results; and easy manual, interactive annotation in 3D.
```
Comment (psobolewskiPhD) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
At its core, it provides critical viewer features out-of-the-box, such as support for [large multi-dimensional data](dask); [“layers”](layers-glance) to simultaneously visualize images, models, and analysis results; and easy manual, interactive annotation in 3D.
```
I think linking to the layers at a glance is better here?

**Review by psobolewskiPhD — APPROVED** [review]
I went through everything and it's great!
Just a couple small nits...

...plus merge conflicts 😬

**Comment by lucyleeow** [comment]
> ...plus merge conflicts 😬

Oooff I just realised, let me have a look. I am worried the merge conflicts are too complex after my extending napari PR re-organise. I'm tempted to abandon this PR, WDYT?

File: docs/guides/contexts_expressions.md (hunk 2 above)
Comment (lucyleeow) [reply docs/guides/contexts_expressions.md]: Thanks, fixed

File: docs/guides/layers.md (hunk 3 above)
Comment (lucyleeow) [reply docs/guides/layers.md]: Done

**Comment by lucyleeow** [comment]
Okay the conflicts were not as bad as I imagined. I was worried because I had moved files around in other PRs that have now been merged.

@psobolewskiPhD thanks for the review, changes made! (if possible would be great to get this in to avoid more merge conflicts!)

**Comment by psobolewskiPhD** [comment]
Thanks for your efforts @lucyleeow 
I've marked `ready for merge` and will merge tomorrow--if I don't, please ping me!

**Comment by lucyleeow** [comment]
@psobolewskiPhD ping, can this go in? Thank you :pray:

**Comment by psobolewskiPhD** [comment]
Done--sorry & thanks.

**Pull request merged**